roles/openshift_router/tasks/main.yml

```diff
@@ -1,10 +1,66 @@
 ---
-- name: Deploy OpenShift Router
+- fail:
+    msg: "Both 'certfile' and 'keyfile' keys must be specified when supplying the openshift_router_certificate variable."
+  when: openshift_router_certificate is defined and ('certfile' not in openshift_router_certificate or 'keyfile' not in openshift_router_certificate)
+
+- name: Set router facts
+  openshift_facts:
+    role: router
+    local_facts:
+      selector: "{{ openshift_router_selector | default(openshift_infra_selector | default(None)) }}"
+
+- name: Deploy router certificate and key
+  copy:
+    src: "{{ item }}"
+    dest: "{{ openshift_master_config_dir }}"
+    mode: 0600
+  with_items:
+  - "{{ openshift_router_certificate.certfile }}"
+  - "{{ openshift_router_certificate.keyfile }}"
+  when: openshift_router_certificate is defined
+
+- name: Read certificates
+  slurp:
+    src: "{{ item }}"
+  register: openshift_router_certificate_output
+  with_items:
+  - "{{ openshift_master_config_dir }}/{{ openshift_router_certificate.certfile | basename }}"
+  - "{{ openshift_master_config_dir }}/{{ openshift_router_certificate.keyfile | basename }}"
+  when: openshift_router_certificate is defined
+
+- name: Create PEM certificate
+  template:
+    src: openshift-router.pem.j2
+    dest: "{{ openshift_master_config_dir }}/{{ openshift_router_certificate.certfile | basename }}.pem"
+    mode: 0600
+  when: openshift_router_certificate is defined
+
+- name: Retrieve list of openshift nodes
+  command: >
+    {{ openshift.common.client_binary }} --api-version='v1' -o json
+    get nodes -n default --config={{ openshift.common.config_base }}/master/admin.kubeconfig
+  register: openshift_router_nodes_json
+
+- name: Collect nodes matching router selector
+  set_fact:
+    openshift_router_nodes: >
+      {{ (openshift_router_nodes_json.stdout|from_json)['items']
+         | oo_oc_nodes_matching_selector(openshift.router.selector) }}
```

Comment:
What if we have an optional variable to pass in for a list of nodes? That would allow for initial installation to still run this role prior to nodes being installed.